src/mainwindow.cpp

```diff
@@ -1519,3 +1520,45 @@ void MainWindow::resetFilters()
     m_hashTableProxy->setShowCheckedRowsOnly(false,false);
     m_hashTableProxy->setFilterRegExp("");
 }
+
+void MainWindow::getDefaultFormat()
+{
+    // This signal may be called 1 ms before applySettings() so use the lineEdit text
+    m_johnDefaultFormat.setJohnProgram(m_ui->lineEditPathToJohn->text());
+    QStringList args;
+    args << "-stdin";
+    args << "--session=defaultFormat";
```

Comment:
In any case, you should put a path that honors ~/.john/sessions instead of polluting JtR's run dir.